### PR TITLE
Lots of species stuff ported from Baystation

### PR DESCRIPTION
**Pull request by BothNevarBackwards**
A bunch of stuff that lets species have more custom behaviours.

Edit: the footprints are the only thing that has an impact without adding more code.

### CODE REVIEW

**Comment by Anewbe** [comment]
Have you tested all of this?

**Comment by BothNevarBackwards** [comment]
Yup, I tested each between commits.

**Comment by Anewbe** [comment]
On our code, or on Baycode?
Because I don't actually remember getting the flash thing to work.

**Comment by BothNevarBackwards** [comment]
I tested it locally. I assume that it has been tested for baycode but all testing I did was with polaris code.

File: code/game/objects/effects/decals/Cleanable/tracks.dm [hunk 1]
```diff
@@ -154,20 +154,49 @@ var/global/list/image/fluidtrack_cache=list()
...
+	drydesc = "They look like dried tracks left by a mammal."
+	coming_state = "paw1"
+	going_state  = "paw2"
+	random_icon_states = null
+	amount = 0
+
+/obj/effect/decal/cleanable/blood/tracks/claw
+	name = "wet tracks"
+	dryname = "dried tracks"
+	desc = "They look like still wet tracks left by a reptile."
```
Comment (Belsima) [inline code/game/objects/effects/decals/Cleanable/tracks.dm]: Should probably generalize it to a clawed creature. Non-reptiles can easily have similiar feet e.g. avians.

**Comment by Neerti** [comment]
Can't see any problem with this.

**Pull request merged**